torch/jit/__init__.py

```diff
@@ -1613,27 +1638,51 @@ def _get_weak_stubs(cls):
     return stubs
 
 
-def _make_strong(mod):
+def _convert_to_script_module(mod, methods=None):
     """
-    Converts a weak module into a subclass of ScriptModule
+    Makes a ScriptModule from an nn.Module. If `_methods` is provided,
+    these methods are treated as @script_methods. If not, it defaults to
+    `('forward',)`. Methods accessed in forward are scripted on demand if
+    `_enable_recursive_script()` is used.
     """
+    if methods is None:
+        methods = ('forward',)
+    exported = ()
+    if hasattr(mod, '__torchscript_export__'):
+        exported = tuple(mod.__torchscript_export__)
+    methods = methods + exported
+
     if mod in _jit_internal.weak_modules:
```

Comment:
It is possible we no longer want to cache this.    Should calling script twice on the same nn.Module return the same module? What if you mutate something in the original class (e.g. do some model surgery) and recall it?

Reply:
I agree, it should make a fresh copy each time to stay with the idea of it being a "pure" operation. I think that should be in a follow up diff though.